### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mrogozhinskaya**
- [DEMO LINK](https://mrogozhinskaya.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://mrogozhinskaya.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/styles/index.scss [hunk 1]
```diff
@@ -1,3 +1,147 @@
...
+  $color-white: #fff;
+  $gap: 4px;
+  
+  body,
+  p,
+  a,
+  img,
+  div {
+    margin: 0;
+    box-sizing: border-box;
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:15]: would be better to set this property via * selector (only box-sizing)

File: src/styles/index.scss [hunk 2]
```diff
@@ -1,3 +1,147 @@
...
+    font-family: Roboto, Helvetica, sans-serif;
+  }
+  
+  .card {
+    width: 200px;
+    height: 406px;
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:21]: do not hardcode height. just set proper paddings and margins everywhere

File: src/styles/index.scss [hunk 3]
```diff
@@ -1,3 +1,147 @@
...
+      line-height: 18px;
+      font-weight: 500;
+      color: $color-black;
+      overflow: hidden;
+    }
+  
+    &__code {
+      margin-top: 4px;
+      font-size: 10px;
+      font-style: normal;
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:47]: this is default value you don't need to set it explicitly. Check everywhere

File: src/styles/index.scss [hunk 4]
```diff
@@ -1,3 +1,147 @@
...
+      font-weight: 400;
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:48]: set font-weight for the entire page by default and only override when needed.

File: src/styles/index.scss [hunk 5]
```diff
@@ -1,3 +1,147 @@
...
+    &__reviews {
+      font-size: 10px;
+      font-style: normal;
+      font-weight: 400;
+    }
+  
+    &__cost {
+      display: flex;
+      justify-content: space-between;
+      height: 18px;
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:69]: same. don't hardcode height

File: src/styles/index.scss [hunk 6]
```diff
@@ -1,3 +1,147 @@
...
+      font-size: 16px;
+      font-weight: 700;
+      line-height: 18px;
+      color: $color-black;
+    }
+  
+    &__link {
+      display: flex;
+      width: 100%;
+      height: 40px;
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:91]: the same here

File: src/styles/index.scss [hunk 7]
```diff
@@ -1,3 +1,147 @@
...
+      background-image: url(../images/star-active.svg);
+    }
+  
+    &--4 > :nth-child(-n+4) {
+      background-image: url(../images/star-active.svg);
+    }
+  
+    &--5 > :nth-child(-n+5) {
+      background-image: url(../images/star-active.svg);
+    }
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:141]: combine these selectors with coma

File: src/styles/mixins.scss [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+@mixin with-centered-content {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
```
Comment (anastasiia-tilikina) [inline src/styles/mixins.scss:16]: you don't use this in your code. So either use them or delete.

**Review by anastasiia-tilikina — CHANGES_REQUESTED** [review]
Nice start!

Yet we have some things to fix.

File: src/styles/index.scss (hunk 1 above)
Comment (mrogozhinskaya) [reply src/styles/index.scss:15]: не зовсім розумію. Тобто видалити margin ?

File: src/styles/index.scss (hunk 6 above)
Comment (mrogozhinskaya) [reply src/styles/index.scss:91]: Яким чином необхідно указати висоту ?

File: src/styles/index.scss (hunk 1 above)
Comment (IvanFesenko) [inline src/styles/index.scss:15]:
```suggestion
* {
  box-sizing: border-box;
}

  body,
  p,
  a,
  img,
  div {
    margin: 0;
```

File: src/styles/index.scss (hunk 2 above)
Comment (IvanFesenko) [reply src/styles/index.scss:21]: not fixed
```suggestion


```
2023-10-03_13h31_00

File: src/styles/index.scss [hunk 9]
```diff
@@ -1,3 +1,147 @@
...
+      display: block;
+      width: 160px;
+      height: 134px;
+      margin: 0 auto;
+    }
+  
+    &__title {
+      margin-top: 40px;
+      margin-bottom: 0;
+      font-style: normal;
```
Comment (IvanFesenko) [inline src/styles/index.scss:36]: `font-style` by default is normal - redundant. Fix all cases

File: src/styles/index.scss (hunk 4 above)
Comment (IvanFesenko) [inline src/styles/index.scss:48]: Not fixed 
2023-10-03_13h32_06

File: src/style.css [hunk 10]
```diff
@@ -1,3 +1,134 @@
...
 body {
   margin: 0;
 }
+
+p {
+  padding: 0;
+  margin: 0;
+}
+
+.card {
```
Comment (Viktor-Kost) [inline src/style.css:19]: why do you duplicate styles for .card and its elements in styles.css when you have those styles at card.scss? just import card.scss. The same with stars.scss

File: src/index.html [hunk 11]
```diff
@@ -3,11 +3,51 @@
...
     <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <a href="#">
+        <img
+          src="images/imac.jpeg"
+          alt="APPLE A1419 iMac image"
+          class="card__img"
+        >
+      </a>
+      <a href="#" class="card__title">
```
Comment (Viktor-Kost) [inline src/index.html:24]: it's not a link
```suggestion
      <h2 class="card__title">
```

File: src/index.html [hunk 12]
```diff
@@ -3,11 +3,51 @@
...
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+        </div>
+
+        <p class="card__reviews">Reviews: 5</p>
+      </div>
+      <div class="card__bottom">
+        <p class="card__bottom_text">Price:</p>
+        <p class="card__bottom_price">$2,199</p>
+      </div>
```
Comment (Viktor-Kost) [inline src/index.html:43]:
```suggestion
      <div class="card__price">
        <p class="card__price_text">Price:</p>
        <p class="card__price_value">$2,199</p>
      </div>
```

File: src/index.html [hunk 13]
```diff
@@ -3,11 +3,51 @@
...
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;600;700&family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+    <title>Card</title>
     <link rel="stylesheet" href="./style.css">
   </head>
   <body>
     <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <a href="#">
```
Comment (Viktor-Kost) [inline src/index.html:17]: this link wrapper is redundant
```suggestion
```

File: src/card.scss [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+      color: $second-main-color;
+    }
+    &_price {
+      font-weight: 700;
+      line-height: 18px;
+      color: $main-text-color;
+    }
+  }
+  &__button {
+    height: 40px;
```
Comment (Viktor-Kost) [inline src/card.scss:68]: it's not a good practice to hardcode height for elements with content inside. Just use padding

File: src/card.scss [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+    display: flex;
+    align-content: center;
+  }
+  &__reviews {
+    font-weight: normal;
+    font-size: 10px;
+    color: $main-text-color;
+    text-align: center;
+  }
+  &__bottom {
```
Comment (Viktor-Kost) [inline src/card.scss:50]:
```suggestion
  &__price {
```

File: src/card.scss [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: 16px;
+    &_text {
+      font-weight: normal;
+      font-size: 12px;
+      line-height: 18px;
+      color: $second-main-color;
+    }
+    &_price {
```
Comment (Viktor-Kost) [inline src/card.scss:61]:
```suggestion
    &_value {
```

File: src/card.scss [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+    color: $main-text-color;
+    text-align: center;
+  }
+  &__bottom {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: 16px;
+    &_text {
+      font-weight: normal;
```
Comment (Viktor-Kost) [inline src/card.scss:56]:
```suggestion
```

File: src/card.scss [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+    justify-content: space-between;
+    align-items: center;
+    margin-bottom: 24px;
+  }
+  &__stars {
+    display: flex;
+    align-content: center;
+  }
+  &__reviews {
+    font-weight: normal;
```
Comment (Viktor-Kost) [inline src/card.scss:45]:
```suggestion
```

File: src/card.scss [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+    margin: 40px 0 4px;
+    text-decoration: none;
+    font-weight: 500;
+    font-size: 12px;
+    line-height: 18px;
+    color: $main-text-color;
+  }
+  &__code {
+    margin-bottom: 16px;
+    font-weight: normal;
```
Comment (Viktor-Kost) [inline src/card.scss:29]:
```suggestion
```

File: src/card.scss [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+  justify-content: center;
+  padding: 32px 16px 16px;
+  max-width: 200px;
+  border: 1px solid $main-border-color;
+  border-radius: 5px;
+  &__img {
+    display: block;
+    max-width: 160px;
+    height: 134px;
+    margin: 0 auto;
```
Comment (Viktor-Kost) [inline src/card.scss:17]:
```suggestion
    margin: 0 auto 40px;
```

File: src/card.scss [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+  }
+  &__title {
+    margin: 40px 0 4px;
```
Comment (Viktor-Kost) [inline src/card.scss:20]:
```suggestion
    margin-bottom: 4px;
```

File: src/stars.scss [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+  background-position: center;
+  background-repeat: no-repeat;
+
+  &:nth-child(-n + 4) {
+    background-image: url(../images/star-active.svg);
+  }
+
+  &--5 {
+    background-image: url(../images/star.svg);
+  }
```
Comment (Viktor-Kost) [inline src/stars.scss:16]: apply styles for all modifiers (1-5). Just reuse your solution from a previous task

File: src/stars.scss [hunk 23]
```diff
@@ -0,0 +1,17 @@
+.stars {
+  display: flex;
+  height: 16px;
+  width: 16px;
+  margin-right: 4px;
```
Comment (Viktor-Kost) [inline src/stars.scss:5]: last child shouldn't have margin. use `:not(:last-child)`

File: src/stars.scss [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+  cursor: pointer;
```
Comment (Viktor-Kost) [inline src/stars.scss:6]:
```suggestion
```

File: src/stars.scss [hunk 25]
```diff
@@ -0,0 +1,17 @@
+.stars {
+  display: flex;
```
Comment (Viktor-Kost) [inline src/stars.scss:2]: + add styles for `.stars`
```suggestion
&__star {
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/style.css (hunk 10 above)
Comment (mrogozhinskaya) [reply src/style.css:19]: Не вийшло імпортувати стилі  card.scss, видає помилку. 
https://media.mate.academy/chats/20607/c9561ee2-a93d-497d-be1a-262fa8281b09.png
https://media.mate.academy/chats/20607/dd7d3685-76b9-45a8-80c6-7958e1131152.png

Comment (VitaliyBodnar) [reply src/style.css:19]: please ask this question in fe_chat

Comment (VitaliyBodnar) [reply src/style.css:19]: after that delete duplicated styles from styles.css

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**